Modules/xASL_module_ASL.m

```diff
@@ -224,19 +185,88 @@
 if isfield(x.modules.asl,'TimeEncodedMatrixType') || isfield(x.modules.asl,'TimeEncodedMatrixSize')
     x.modules.asl.bTimeEncoded = 1;
 end
-
 % Check if there is Decoding Matrix as input
 %(some datasets will have a decoding matrix that we can use directly in the decoding part)
 
 
-%% Multi-TE parsing
+%% D5. Multi-TE parsing
+
+
+
+%% E1. Default quantification parameters in the Q field
+if ~isfield(x,'Q')
+    x.Q = struct;
+end
+
+if ~isfield(x.Q,'ApplyQuantification') || isempty(x.Q.ApplyQuantification)
+    x.Q.ApplyQuantification = [1 1 1 1 1]; % by default we perform scaling/quantification in all steps
+elseif length(x.Q.ApplyQuantification)>5
+    warning('x.Q.ApplyQuantification had too many parameters');
+    x.Q.ApplyQuantification = x.Q.ApplyQuantification(1:5);
+elseif length(x.Q.ApplyQuantification)<5
+    warning('x.Q.ApplyQuantification had too few parameters, using default 1');
+    x.Q.ApplyQuantification(length(x.Q.ApplyQuantification)+1:5) = 1;
+end
+
+if ~isfield(x.Q,'BackgroundSuppressionNumberPulses') && isfield(x,'BackgroundSuppressionNumberPulses')
+    % Temporary backwards compatibility that needs to go
+    x.Q.BackgroundSuppressionNumberPulses = x.BackgroundSuppressionNumberPulses;
+end
+
+if ~isfield(x.Q, 'bUseBasilQuantification') || isempty(x.Q.bUseBasilQuantification)
+	x.Q.bUseBasilQuantification = false;
+end
 
 
-%% Define sequence (educated guess)
+%% E2. Define sequence (educated guess based on the Q field)
 x = xASL_adm_DefineASLSequence(x);
 
 
-%% Skip processing if invalid image
+%% F. Backward and forward compatibility of filenames
+if ~xASL_exist(x.P.Path_M0, 'file')
```

Comment:
You have reverted the order of the code for M0-path fixing and ASL-M0 splitting. This in theory should be fine - but in theory only ;)

Reply:
True, that's why I tested the testdatasets